setup/git.md

```diff
@@ -0,0 +1,23 @@
+# Git and Github
+
+## Git
+
+Git is a version control software, which allows you to save your code in steps, like different drafts for an essay. It also allows for branches, which could represent someone else's edits to your paper, which then you merge into your main paper.
+
+For this tutorial, git and Github are not nessicary, but highly recommended. You can download git from: https://git-scm.com/.
```

Comment:
Here are a couple more git topics we may want actually. Feel free to knock down some of these though as they may be outside the scope of a modder's intro to modding Minecraft rather than a full git tutorial.
1. Setting your global/project git credentials
2. what exactly a commit is
3. how to handle merge conflicts
4. the push/pull system
5. rebase vs merge commits
6. branches (and when to use them)
7. FOSS and why you should be